tests/xharness/AppRunner.cs

```diff
@@ -326,168 +326,228 @@ public async Task<ProcessExecutionResult> UninstallAsync ()
-
-			// parse the xml and build a human readable version
-			int total = int.Parse (resultsNode.Attributes ["total"].InnerText);
-			int errors = int.Parse (resultsNode.Attributes ["errors"].InnerText);
-			int failed = int.Parse (resultsNode.Attributes ["failures"].InnerText);
-			int notRun = int.Parse (resultsNode.Attributes ["not-run"].InnerText);
-			int inconclusive = int.Parse (resultsNode.Attributes ["inconclusive"].InnerText);
-			int ignored = int.Parse (resultsNode.Attributes ["ignored"].InnerText);
-			int skipped = int.Parse (resultsNode.Attributes ["skipped"].InnerText);
-			int invalid = int.Parse (resultsNode.Attributes ["invalid"].InnerText);
-			int passed = total - errors - failed - notRun - inconclusive - ignored - skipped - invalid;
-			var testFixtures = resultsNode.SelectNodes ("//test-suite[@type='TestFixture' or @type='TestCollection']");
-			for (var i = 0; i < testFixtures.Count; i++) {
-				var node = testFixtures [i];
-				writer.WriteLine (node.Attributes ["name"].InnerText);
-				var testCases = node.SelectNodes ("//test-case");
-				for (var j = 0; j < testCases.Count; j++) {
-					var result = testCases [j];
-					var status = result.Attributes ["result"].InnerText;
-					switch (status) {
-					case "Success":
-						writer.Write ("\t[PASS] ");
-						break;
-					case "Ignored":
-						writer.Write ("\t[IGNORED] ");
-						break;
-					case "Error":
-					case "Failure":
-						writer.Write ("\t[FAIL] ");
-						break;
-					case "Inconclusive":
-						writer.Write ("\t[INCONCLUSIVE] ");
-						break;
-					default:
-						writer.Write ("\t[INFO] ");
-						break;
+			long total, errors, failed, notRun, inconclusive, ignored, skipped, invalid;
+			total = errors = failed = notRun = inconclusive = ignored = skipped = invalid = 0L;
+			using (var reader = XmlReader.Create (stream)) {
+				while (reader.Read ()) {
+					if (reader.NodeType == XmlNodeType.Element && reader.Name == "test-results") {
+						total = long.Parse (reader ["total"]);
+						errors = long.Parse (reader ["errors"]);
+						failed = long.Parse (reader ["failures"]);
+						notRun = long.Parse (reader ["not-run"]);
+						inconclusive = long.Parse (reader ["inconclusive"]);
+						ignored = long.Parse (reader ["ignored"]);
+						skipped = long.Parse (reader ["skipped"]);
+						invalid = long.Parse (reader ["invalid"]);
 					}
-					writer.Write (result.Attributes ["name"].InnerText);
-					if (status == "Failure" || status == "Error") { //  we need to print the message
-						writer.Write ($" : {result.InnerText}");
+					if (reader.NodeType == XmlNodeType.Element && reader.Name == "test-suite" && (reader["type"] == "TestFixture" || reader["type"] == "TestCollection")) {
+						var testCaseName = reader ["name"];
+						writer.WriteLine (testCaseName);
+						var time = "";
+						try {
+							time = reader ["time"];
+						} catch (Exception) { // some nodes might not have the time :/
```

Comment:
Can you check if the node has "time" instead of catching the exception? Exceptions are expensive...

Reply:
Yeah, I did not read all the docs, GetAttribute will do the right thing.